controllers/quay/quayregistry_controller.go

```diff
@@ -348,7 +348,14 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 				if upgradeJob.Status.Succeeded > 0 {
 					log.Info("Quay upgrade complete, updating `status.currentVersion`")
 
-					updatedQuay, _ := v1.EnsureRegistryEndpoint(quayContext, updatedQuay, userProvidedConfig)
+					var freshQuay v1.QuayRegistry
+					if err := r.Client.Get(ctx, req.NamespacedName, &freshQuay); err != nil {
```

Comment:
What's the reasoning for this extra request first, as opposed to using `updatedQuay` directly?

Reply:
If the version we have in memory is not the current version (the version in the etcd) we will receive a Conflict error. If we don't fetch a fresh version, once we receive a Conflict back, we will always see Conflicts.